### DISCUSSION TITLE
Reconsider the name

### DISCUSSION THREAD

**Discussion by BafS** [NativePHP/laravel#342] [Ideas]
The name "NativePHP" is misleading because it uses Electron, which uses the Chromium engine for the UI, which is not native. It's an hybrid or cross-platform application rather than a native one.

I have seen comments about this on reddit more than once (For eg. https://www.reddit.com/r/PHP/comments/1chsxfe/comment/l2712t1) and I agree with them.

Why is it not native?

 - It's not using platform-Specific APIs for the UI. The UI rendering is done by the chromium engine, it's a "canvas" that doesn't use the native API sets (operating system's native toolkits) provided by the operating system
 - Electron apps, due to their web-based nature, don't perform as well compared to true natives apps in terms of speed and resource usage because of the web UI layer
 - Native apps don't have to ship with a rendering engine (chromium in this case) because it's handled by the operating system

Why is React Native named like that if React & electron was native already?

> We partnered with the Messenger team at Facebook to enable user experiences that are only possible on desktop, and to deliver huge performance wins over their previous Electron implementation. Our vision is to combine high quality, delightful, _native_ desktop user experiences with the great developer experience of React Native.

https://reactnative.dev/blog/2021/08/19/h2-2021#messenger-desktop

If you are still not convinced we can use an _argumentum ad absurdum_; let's assume that NativePHP is native, if I write an OS in NativePHP it is, by definition, native. I can then write NativePHP2 in my new OS, which is still native. We can continue to apply this logic forever which leads us down a slippery slope where everything becomes native by definition.

If we apply the term "native" so broadly that even web apps running on users' machines qualify as such, we've essentially rendered the term meaningless. Many projects specifically use the term "native" to show that it's not using electron/web-ui.

Please reconsider the name.

**Comment by simonhamp** [comment]
We won't be changing the name
